src/config.ts

```diff
@@ -0,0 +1,78 @@
+import {ActionInputs} from "./args";
+import fetch from 'node-fetch';
+
+export interface TarpaulinConfig {
+    /**
+     * Additional command line options.
+     */
+    additionalOptions: string[],
+
+    /**
+     * The URL to download a tarball of cargo-tarpaulin from.
+     */
+    downloadUrl: string,
+
+    /**
+     * The type of tests to run. If {@code null}, doctests and normal tests
+     * will be run.
+     */
+    type: string | null,
+
+    /**
+     * The maximum time a test can be ran without response before a timeout occurs.
+     */
+    timeout: string | null,
+}
+
+/**
+ * Resolve the configuration (e.g., download url and test options) required to run tarpaulin
+ * from the inputs supplied to the action.
+ *
+ * @param input The parameters of the action.
+ */
+export default async function resolveConfig(input: ActionInputs): Promise<TarpaulinConfig> {
+    const downloadUrl = await getDownloadUrl(input.releaseEndpoint, input.requestedVersion);
+    const type = input.runType ? input.runType : null;
+    const timeout = input.timeout ? input.timeout : null;
+
+    let additionalOptions: string[] = [];
+
+    if (input.opts !== null) {
+        let opts = input.opts.match(/[^\s]+|"(?:\\"|[^"])+"/g);
```

Comment:
I used [`string-argv`](https://www.npmjs.com/package/string-argv) npm package to parse CLI arguments string in `actions-rs/cargo`, we could use it here too, as it should handle any edge cases better